tests/pstdarg_test.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2017 Jean-Damien Durand <jeandamiendurand@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "plibsys.h"
+#include "ptestmacros.h"
+
+#include <stdio.h>
+#include <stdarg.h>
+#include <string.h>
+
+P_TEST_MODULE_INIT ();
+
+static void f (int i, ...) {
+  p_va_list args1, args2;
+
+  p_va_start (args1, i);
+  p_va_copy(args2, args1);
+
+  P_TEST_CHECK (va_arg (args1, int) == 42);
+  P_TEST_CHECK (va_arg (args2, int) == 42);
+
+  p_va_end (args1);
+  p_va_end (args2);
+}
+
+P_TEST_CASE_BEGIN (pstdarg_general_test)
+{
+	p_libsys_init ();
+
+        f(0 ,42);
```

Comment:
First, could you please give a more meaningful name to the test function? Something like variadic_function(arg_count, ...) or similar?

Second, can we make this test a bit more complicated to catch possible errors in OS-dependent implementations? I mean, to use not only the one parameter, but at least 2 or 3. Just to iterate through all the arguments and check them. This will not only add some additional check that in general case everything should work fine, but also will give a nice example for potential user on how to use this API.

And can we also make tests with different data types, i.e. pint, pint64, float, double?

Reply:
Done